### ISSUE TITLE
multiple openapi.yaml files show errors

### ISSUE THREAD

**Issue by jahluwalia** [Mermade/openapi-lint-vscode#7]
Hi,

I have my spec spread out across multiple *.oas3.yaml files. (e.g. root.oas3.yaml and definitions.oas3.yaml) that use the $ref.

If i "resolve and validate" the root file, I receive a notification that the file is valid. 

However, if i go into definitions.oas3.yaml, I see red underlines all over the place. For example "Unexpected property ErrorResponse" among others. 

How can we let the extension know that this is part of a larger file structure and not show errors when they aren't really errors. 

thanks,
jas

**Comment by MikeRalphson** [comment]
Is your `definitions.oas3.yaml` file a fully valid OpenAPI document, i.e. does it include properties `openapi, info, paths`?

**Comment by MikeRalphson** [comment]
https://dev.to/mikeralphson/defining-reusable-components-with-the-openapi-specification-4077

**Comment by jahluwalia** [comment]
Hi @MikeRalphson 

The article was great and highlighted something for me which was there is no right way of doing things in the spec (so my comment of valid vs invalid probably isn't right). 

I wasn't sure how i could maintain my file structure. So an example is as follows:

root.oas3.yaml:

```
openapi: 3.0.0
info:
  version: 0.0.1
  title: myService
paths:
  /healthcheck:
    $ref: ./healthcheck.oas3.yaml
servers:
  - url: /
```

healthcheck.oas3.yaml:

```
get:
    description: Health check 
    operationId: healthcheck
    x-swagger-router-controller: healthcheck
    responses:
        "200":
            description: Service is healthy and all dependencies are up
            content:
                application/json:
                    schema:
                        type: array
                        items:
                            type: string
        default:
            description: Error
            content:
                application/json:
                    schema:
                        $ref: "./definitions.oas3.yaml#/ErrorResponse"
```

definitions.oas3.yaml:

```
ErrorResponse:
    required:
        - message
    properties:
        message:
            type: "string"
            description: "The description of the error"
        code:
            type: "string"
            description: "Identifies the kind of error"
```

**Comment by MikeRalphson** [comment]
I would *personally* structure your files as follows:

```yaml
openapi: 3.0.0
info:
  version: 0.0.1
  title: myService
paths:
  /healthcheck:
    $ref: "./healthcheck.oas3.yaml#/paths/~1healthcheck"
servers:
  - url: /
```

```yaml
openapi: 3.0.0
info:
  title: Healthcheck fragment
  version: 0.0.1
paths:
  /healthcheck:
    get:
        description: Health check 
        operationId: healthcheck
        x-swagger-router-controller: healthcheck
        responses:
            "200":
                description: Service is healthy and all dependencies are up
                content:
                    application/json:
                        schema:
                            type: array
                            items:
                                type: string
            default:
                description: Error
                content:
                    application/json:
                        schema:
                            $ref: "./definitions.oas3.yaml#/components/schemas/ErrorResponse"
```

```yaml
openapi: 3.0.0
info:
  title: Definitions fragment
  version: 0.0.1
paths: {}
components:
  schemas:
    ErrorResponse:
        required:
            - message
        properties:
            message:
                type: "string"
                description: "The description of the error"
            code:
                type: "string"
                description: "Identifies the kind of error"
```

The `path` in `healthcheck.oas3.yaml` could in fact be anything (including just `/`) but having it be different from the referring `path` key might be confusing.

There is a tracking issue on the OAS spec. repository for [making the paths object optional](https://github.com/OAI/OpenAPI-Specification/pull/1781)  in OAS v3.1.0.

**Comment by MikeRalphson** [comment]
Your examples highlighted a missing option being passed to the resolver, so that's now fixed in `v1.0.1`. Thanks!

**Comment by jahluwalia** [comment]
Glad I was able to help, but not sure how :)

I'm going to try to use your model going forward. Thanks for the advice.

**Comment by MikeRalphson** [comment]
No worries - please feel free to chase up this issue with any further comments. Hopefully it will be helpful to other people too.
